pkg/ipfs/eth_state_trie/node.go

```diff
@@ -0,0 +1,55 @@
+package eth_state_trie
+
+import (
+	"gx/ipfs/QmcZfnkapfECQGcLZaf9B79NRg7cRa9EnZh4LSbkCzwNvY/go-cid"
+	"gx/ipfs/Qme5bWv7wtjUNGsK2BNGVUFPKiuxWrsqrtvYwCLRw8YFES/go-ipld-format"
+)
+
+type EthStateTrieNode struct {
+	cid     *cid.Cid
+	rawdata []byte
+}
+
+func (estn EthStateTrieNode) RawData() []byte {
+	return estn.rawdata
+}
+
+func (estn EthStateTrieNode) Cid() *cid.Cid {
+	return estn.cid
+}
+
+func (EthStateTrieNode) String() string {
```

Comment:
It looks like EthStateTrieNode is implementing ipfs Node & Resolver interfaces. Are there any others that it is implementing to account for `String()`, `Loggable()`, etc)?

Reply:
It's implementing the Node interface from go-ipld-format: https://github.com/ipfs/go-ipld-format/blob/master/format.go#L26

That interface includes the interfaces for Resolver and Block, which I think together account for all of the required methods.

Planning to add actual implementations for those methods if/when we start using them.

Reply:
Gotcha, I missed that it was also implementing the Block interface. 👍 Makes sense.